File: solutions/0020_Valid_Parentheses/stack.py
class Solution:
    def isValid(self, s: str) -> bool:
        checker = {')': '(', ']': '[', '}': '{'}
        stack = []
        for i in range(len(s)):
            c = s[i]
            if c == '(' or c == '[' or c =='{':
                stack.append(s[i])
            else:
                if len(stack) == 0 or stack.pop() != checker[c]:
                    return False
        return len(stack) == 0

File: solutions/0150_Evaluate_Reverse_Polish_Notation/stack.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        for (String token : tokens) {
            if (token.equals("+") || token.equals("-") ||
               token.equals("*") || token.equals("/")) {
                Integer b = stack.pop();
                Integer a = stack.pop();
                if (token.equals("+")) stack.push(a + b);
                if (token.equals("-")) stack.push(a - b);
                if (token.equals("*")) stack.push(a * b);
                if (token.equals("/")) stack.push(a / b);
            } else {
                stack.push(Integer.valueOf(token));
            }
        }
        return stack.pop();
    }
}
